src/box/iproto_constants.h

```diff
@@ -218,7 +218,23 @@ extern const char *iproto_flag_bit_strs[];
 	/**
 	 * Flag indicating whether the transaction is synchronous.
 	 */								\
-	 _(IS_SYNC, 0x61, MP_BOOL)
+	 _(IS_SYNC, 0x61, MP_BOOL)					\
+	/**
+	 * Cursor, used in IRPOTO_FETCH_SNAPSHOT, in order to fetch snapshot
+	 * from a file:
+	 *
+	 * IPROTO_CURSOR: {
+	 *      IRPOTO_ENGINE: number,
+	 *      IPROTO_VCLOCK: map,
+	 *      IPROTO_LSN: number,
```

Comment:
- I don't think you need a vclock to resume - signature would be enough.
- The name "lsn" is confusing - it's a row number in the snapshot and should be named appropriately.
- Why do you need to put these fields in a map? Why not store them at the top level?
- `IPROTO_CURSOR` is a too generic name - for example, it could be used for pagination.

Reply:
> I don't think you need a vclock to resume - signature would be enough.

It'll be used in persistent GC. Moreover, we send VCLOCK during JOIN, it won't be great to expect client to modify the data and send it back to us.

> The name "lsn" is confusing - it's a row number in the snapshot and should be named appropriately.

Let's not create a new iproto constant just for that, we know, that LSN is a number of row in snap.

> Why do you need to put these fields in a map? Why not store them at the top level?

Because in the future there may be other VCLOCK and LSN required for the more general case, than checkpoint cursor. If we put them at the top level, then we'll have to create new iproto constants for the future features, if they'll use VCLOCK and LSN.

> IPROTO_CURSOR is a too generic name - for example, it could be used for pagination.

Sure, let's change it to `IPROTO_CHECKPOINT_CURSOR`

Reply:
> > Why do you need to put these fields in a map? Why not store them at the top level?
> 
> Because in the future there may be other VCLOCK and LSN required for the more general case, than checkpoint cursor. If we put them at the top level, then we'll have to create new iproto constants for the future features, if they'll use VCLOCK and LSN.

Why not just call the new constants `IPROTO_CHECKPOINT_VCLOCK` and `IPROTO_CHECKPOINT_LSN`? Putting them in a map complicates parsing (with a flat structure, you'd just put the new constants in the switch-case in `xrow_decode_replication_request`).

Reply:
> Why not just call the new constants IPROTO_CHECKPOINT_VCLOCK and IPROTO_CHECKPOINT_LSN?

Done